src/modules/cmd_duration.rs

```diff
@@ -64,12 +65,24 @@ fn undistract_me<'a, 'b>(
 fn undistract_me<'a, 'b>(
     module: Module<'a>,
     config: &'b CmdDurationConfig,
+    context: &'a Context,
     elapsed: u128,
 ) -> Module<'a> {
     use notify_rust::{Notification, Timeout};
     use nu_ansi_term::{unstyle, AnsiStrings};
 
     if config.show_notifications && config.min_time_to_notify as u128 <= elapsed {
+        if cfg!(target_os = "linux") {
+            let in_graphical_session = ["DISPLAY", "WAYLAND_DISPLAY", "MIR_SOCKET"]
+                .iter()
+                .map(|&var| context.get_env(var).filter(|val| !val.is_empty()))
```

Comment:
Not a blocker but I think you can use a [find_map](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find_map) here to simplify this a bit.

Reply:
thanks for the heads-up, done!

Reply:
for some reason the commit msg broke, but i meant 'refactor block to use `find_map`'